src/cswinrt/strings/WinRT.cs

```diff
@@ -42,34 +54,94 @@ internal sealed class Platform
         internal static extern bool FreeLibrary(IntPtr moduleHandle);
 
         [DllImport("kernel32.dll", EntryPoint = "GetProcAddress", SetLastError = true, BestFitMapping = false)]
-        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, [MarshalAs(UnmanagedType.LPStr)] string functionName);
+        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, sbyte* functionName);
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, ReadOnlySpan<byte> functionName)
+        {
+            fixed (byte* lpFunctionName = functionName)
+            {
+                return TryGetProcAddress(moduleHandle, (sbyte*)lpFunctionName);
+            }
+        }
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, string functionName)
```

Comment:
If this overload is only for downlevel, we should conditionally only include this when the other one is not.
We shouldn't just always include both of these, as only one of them is used at a time.

Reply:
So we should have like two P/Invoke declarations for `GetProcAddress`? One blittable and the other not?

Reply:
I think we should evaluate with @manodasanW what approach to take here with the constraints we have, and possibly reevaluate some of them. For instance, I'm not sure the split for C# 11 is worth the complexity. If the embedded source files have to support downlevel to .NET Standard 2.0, then their current C# 9 language level is already officially not supported anyway. So at that point why not just bump that to C# 11 directly and say if you want to embed CsWinRT, you need to set your language version to C# 11. They'd have to do the same anyway for C# 9, and both are equally "not supported". Doing this would at least let us stop worrying about the language, and just check the TFM.

Reply:
Well the TFM check isn't really needed (after all, UTF-8 string literals are only language-dependent rather than runtime-dependent). It's mostly to make sure that we are on the "correct" language version for the feature (since .NET 7 uses C#11 by default). Actually checking for the C# version is... not a very easy task to do in MSBuild.

Reply:
Yes but my point is even if we kept the limit of C# 9, that'd still be technically unsupported already. So what I'm saying is maybe we might as well just say you need C# 11 for embed support, and make the code simpler and more consistent. I can't make a decision on this though, I'm just suggesting what I think a possible approach could be 🙂

Reply:
We have folks with embedded projections and regular projections that are going to be on the .NET 6 SDK due to that is the LTS build.  So that is why we will still need the non C# 11 fallback.

Reply:
Is this something we could reevaluate after .NET 8 hits RTM? As in, could we just say consumers of the embedded projections would be expected to be using a language version matching whatever the latest LTS is?

Reply:
Yea we will revisit this a couple months after .NET 8 comes out.